### ISSUE TITLE
Timezone issue in bookings

### ISSUE THREAD

**Issue by enricodente** [alextselegidis/easyappointments#910]
Hi Alex and thanks for your work.
I am testing out the platform, and I have experienced an issue, just with a basic booking test.

My server is correctly set to Europe/Rome timezone (php.ini too). Same holds for Admin and Provider users, both set to Rome (+1).
I set the business hours as starting at 9AM.

This is what I tested:
1) I load the booking form, I fill in a test booking using. In this form, by default it will auto-select Berlin (+1), but I have also tested setting Rome (+1) with the same result.
2) I book a slot for 9AM, everything is confirmed, and if I check in the backend I find a booking for 8AM.
If I try to book again, 9AM slot is still free.
3) I book a slot for 10AM, same day, it is registered at 9AM. If I try to book again using the form, now the 9AM slot is not shown.

Basically there is a misalignment of 1h between what's registered and what's shown, but there is no place where I have the timezone set to something different.

Available for any check or test to improve the software and solve the issue, thanks!

**Comment by wvdvegt** [comment]
Probably the issue i reported a number of times #872 (if the provider timezone does not match the client one, wrong slots are removed from display)

**Comment by enricodente** [comment]
I don't think it is the same issue because:

1.  The timezone of provider matches the client's timezone
2.  The slot removed from display is RIGHT, because it is coherent with the slot booked. The problem is that the booked slot is the wrong one. For the example above: client books slot at 10AM, gets registred as 9AM, next client sees the 9AM slot not available.

**Comment by wvdvegt** [comment]
There are at least 4 timezones involved:

1) The database has one
2) PHP's (if not overruled by the application)
3) The provider one
4) The customer one.

The symptoms are very much alike what I noticed when setting the provider TZ to Azores and the customer TZ to Amsterdam. I'll recheck my setup if it's still present in the latest prerelease version

**Comment by alextselegidis** [comment]
Hello! 

Thanks for testing the app and helping making it more stable. 

This is an issue I fixed yesterday, as the booking was registered in the displayed time, but not the actual provider time, which lead to invalid hours being removed from the schedule. 

Please try again with the beta package and let me know if you experience similar problems.

<small>
<table>
<tr>
<td><img src="https://easyappointments.org/img/easyappointments-signature.png"></td>
<td>&nbsp;&nbsp;Alex Tselegidis, Easy!Appointments Creator <br>
&nbsp;&nbsp;<a href="https://alextselegidis.com">Need a customization? Contact me in person!</a></td>
</tr>
</table>
</small>

**Comment by enricodente** [comment]
Thanks Alex, I'll definetely test it out. What is the recommended way of updating without losing config and database settings? A pull from git?

BTW: my installation package was pulled from git around 1h before my issue post, so 17h ago. Your last commit right now is shown as 22h ago, so I believe I was already using the lastest version.

**Comment by wvdvegt** [comment]
I upgraded and my TZ issues are mostly solved (filed a new issue for the customer notification mail showing the provider's TZ).

**Comment by enricodente** [comment]
I confirm I was already running with the latest comment.

I have a big question: is the software taking into account Daylight Time?
This might be the issue here, as my timezone (Europe/Rome) is officially CET+1, but right now, we are still running CET+2 because of Daylight Time, that will end this Sunday.

So I guess that if this is the issue, the client-side is showing time according to CET+2 (correctly), while server-side, everything is registered as CET+1, leading to the 1h difference.

Let me know what you think about this hypotesys. I will check on Monday anyways to have one more test done.

**Comment by ketominer** [comment]
Hi, also experiencing issues with DST here... is there at least a way to lock the full program to only use UTC?

**Comment by lucilanga** [comment]
Hey @ketominer,

in assets/js/frontend_book.js

change:
`
        $('#select-timezone').val(Intl.DateTimeFormat().resolvedOptions().timeZone);
`
to:
 `       $('#select-timezone').val('UTC');
`

This will fix the timezone on the frontend for bookings.
You can change provider's TZ on the backend.
